lib/credentials/shared_ini_file_credentials.js

```diff
@@ -175,6 +180,33 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
     }
   },
 
+  /**
+  * Executes the credential_process and retrieves
+  * credentials from the output
+  * @api private
+  * @param profile [map] credentials profile
+  * @throws SharedIniFileCredentialsProviderFailure
+  */
+  loadViaCredentialProcess: function loadCredentialProcess(profile, callback) {
+    try {
+      var output = proc.execSync(profile['credential_process']);
```

Comment:
I know that some of the existing loaders are required to remain synchronous to avoid breaking existing applications, but since this is a new loader, is there a reason to make it synchronous? The SharedIniFileCredentials already loads asynchronously if the selected profile is for assume role credentials.